ivy/array/utilities.py

```diff
@@ -0,0 +1,116 @@
+# global
```

Comment:
These changes have nothing to do with `result_type`. This should not be included in your PR. If you need these functions/class, see `ivy/array/utility.py`